chia/simulator/block_tools.py

```diff
@@ -414,6 +433,72 @@ def setup_new_gen(
                 removals,
                 cost,
             )
+        if include_transactions and block_fillrate is not None:
+            # if the caller passed in a fill rate, we want to fill blocks up to that fill percentage
+            # we also use BlockBuilder to compress these transactions as well.
+            assert wallet is not None
+            assert rng is not None
+            assert curr.height >= self.constants.HARD_FORK_HEIGHT  # we need new compression for BlockBuilder
+            # function constants
+            adjusted_max_cost: uint64 = uint64(self.constants.MAX_BLOCK_COST_CLVM * block_fillrate / 100)
+            static_cost: uint64 = uint64(4839648)  # cond + exec cost per sb
+            static_comp_cost: uint64 = uint64(7684000)  # byte cost per sb after compression
```

Comment:
this constant looks really suspicious. `BlockBuilder` will tell what the current block cost is (as long as the cost passed to `add_spend_bundles()` is correct).